include/extractor/edge_based_graph_factory.hpp

```diff
@@ -123,8 +129,41 @@ class EdgeBasedGraphFactory
 
     void FlushVectorToStream(std::ofstream &edge_data_file,
                              std::vector<OriginalEdgeData> &original_edge_data_vector) const;
+
+    struct TurnCandidate
+    {
+        EdgeID eid; // the id of the arc
+        bool valid; // a turn may be relevant to good instructions, even if we cannot take the road
+        double angle;                // the approximated angle of the turn
+        TurnInstruction instruction; // a proposed instruction
+        double confidence;           // how close to the border is the turn?
+
+        std::string toString() const
+        {
+            std::ostringstream oss;
+            oss << "[turn] " << eid << " valid: " << valid << " angle: " << angle
+                << " Instruction: " << static_cast<int>(instruction)
+                << " Confidence: " << confidence;
+            return oss.str();
+        }
+    };
+
+    // Use In Order to generate base turns
+    std::vector<TurnCandidate>
+    getTurnCandidates(NodeID from, EdgeID via_edge); // cannot be const due to the counters...
```

Comment:
Is this clang-formatted? Maybe run it on all of the files you touched (`git diff --name-only`).